libutils/file_lib.c

```diff
@@ -1563,6 +1563,28 @@ int ExclusiveFileLock(FileLock *lock, bool wait)
     return LockFD(lock->fd, F_WRLCK, wait);
 }
 
+int ExclusiveFileLockPath(FileLock *lock, const char *fpath, bool wait)
+{
+    assert(lock != NULL);
+    assert(lock->fd < 0);
```

Comment:
There's quite a few places where you assign `lock->fd` to -1 or compare it to 0. I would add a comment or use the macro (`EMPTY_FILE_LOCK`) to make it more obvious what this means.

Reply:
It's an invalid file descriptor, what's pretty clear :wink: Yeah, I'll add a note that it's used as an indicator if the lock is initialized or not.